.github/workflows/build-containers.yml

```diff
@@ -75,120 +75,192 @@ jobs:
+          base_repo=$(echo $config | jq -r '.base_repo')
+          for base_name in $(echo $base_names | tr "," "\n"); do
+            for base_version in $(echo $base_versions | tr "," "\n"); do
+              echo "{ \"image\": \"${{ matrix.image }}\", \"base_name\": \"$base_name\", \"base_version\": \"$base_version\", \"base_repo\": \"$base_repo\" }" >> matrix.json
+            done
+          done
+          cat matrix.json
+          matrix=$(jq -c -s '.' matrix.json)
+          echo "image_matrix=$matrix" >> $GITHUB_OUTPUT
+  
+      - name: Create build matrix
+        id: create-matrix
+        if: steps.check-jsonfile.outputs.config_exists == 'true'
+        run: |
+          matrix=$(jq -c -s '.' matrix.json)
+          echo "image_matrix=$matrix" >> $GITHUB_OUTPUT
+  
+      - name: Read default-build-config.json
+        id: read-default-config
+        if: steps.check-jsonfile.outputs.config_exists == 'false'
+        run: |
+          default_config=$(cat opensciencegrid/default-build-config.json)
+          echo "default-build-config.json content: $default_config"
+          base_names=$(echo $default_config | jq -r '.base_os.base_name | join(",")')
+          base_versions=$(echo $default_config | jq -r '.base_os.base_version | join(",")')
+          base_repo=$(echo $default_config | jq -r '.base_repo')
+          for base_name in $(echo $base_names | tr "," "\n"); do
+            for base_version in $(echo $base_versions | tr "," "\n"); do
+              echo "{ \"image\": \"${{ matrix.image }}\", \"base_name\": \"$base_name\", \"base_version\": \"$base_version\", \"base_repo\": \"$base_repo\" }" >> matrix_default.json
+            done
+          done
+          cat matrix_default.json
+          matrix_default=$(jq -c -s '.' matrix_default.json)
+          echo "default_image_matrix=$matrix_default" >> $GITHUB_OUTPUT
+  
+      - name: Create default build matrix
+        id: create-default-matrix
+        if: steps.check-jsonfile.outputs.config_exists == 'false'
+        run: |
+          matrix_default=$(jq -c -s '.' matrix_default.json)
+          echo "default_image_matrix=$matrix_default" >> $GITHUB_OUTPUT
+  
+  combine-matrices:
     runs-on: ubuntu-latest
-    if: >-
-      github.ref == 'refs/heads/main' &&
-      github.event_name != 'pull_request' &&
-      github.repository_owner == 'opensciencegrid' &&
-      needs.build-image-list.outputs.images != '["dummy"]'
-    needs: [make-date-tag, build-image-list, build]
+    needs: check-config
+    outputs:
+      combined_matrix: ${{ steps.combine.outputs.combined_matrix }}
+    steps:
+      - name: Combine matrices
+        id: combine
+        run: |
+          image_matrix='${{ needs.check-config.outputs.image_matrix }}'
+          default_image_matrix='${{ needs.check-config.outputs.default_image_matrix }}'
+          combined_matrix=$(jq -c -s '.[0] + .[1]' <<<"$image_matrix $default_image_matrix")
```

Comment:
This isn't really going to work when we have multiple images using default or custom images as you only have a single output variable. Skimming https://github.com/actions/runner/pull/2477, it doesn't look like we can really get reliable output from a matrix so I think you'll need to consolidate this into the `build-image-list` single job